lib/util/cache/repository/impl/client/cache-client-factory.ts

```diff
@@ -0,0 +1,36 @@
+import { GlobalConfig } from '../../../../../config/global';
+import type { RepositoryCacheType } from '../../../../../config/types';
+import { logger } from '../../../../../logger';
+import type { CacheClient } from '../../types';
+import { LocalRepoCache } from './local';
+
+export class CacheClientFactory {
+  private static client: CacheClient | null;
+
+  static get(repository: string, cacheType: RepositoryCacheType): CacheClient {
+    if (this.client) {
+      return this.client;
+    }
+
+    const platform = GlobalConfig.get('platform');
+
+    switch (cacheType) {
+      case 'local':
+        this.client = new LocalRepoCache(platform!, repository);
+        break;
+      // istanbul ignore next: untestable
+      default:
+        this.client = new LocalRepoCache(platform!, repository);
+        logger.info(
+          `Repository cache type not supported using type "local" instead`
+        );
+        break;
+    }
+
+    return this.client;
+  }
+
+  static reset(): void {
+    this.client = null;
+  }
```

Comment:
why is this needed for the client ?

Reply:
to change clients I guess and testing. the factory is static, so we will not be able to change clients in such cases.